### ISSUE TITLE
Cannot create from the client

### ISSUE THREAD

**Issue by brianedmunds** [ga-wdi-boston/full-stack-project#770]
I have an issue that I have been thrashing on for a while.

My app is a workout app and it consist of two tables: users and workouts.

The workouts table consists of three columns: id, plan, user_id

The following curl request to heroku works successfully to add a new row to the workouts table:

curl --include --request POST https://infinite-garden-87904.herokuapp.com/workouts --include --request POST --header "Content-Type: application/json" --data '{
"workout": {
"plan": "pushups",
"user_id": 6
}
}'

Curl requests to localhost work as well:

curl http://localhost:4741/workouts --include --request POST --header "Content-Type: application/json" --data '{
"workout": {
"plan": "Pushups",
"user_id": 6
}
}'

However, when I attempt to add to the workouts table from the client, it fails.

Within the browser, in Network, I see the following:
"POST http://localhost:4741/workouts 400 (Bad Request)
XHR finished loading: POST "http://localhost:4741/workouts".

{status: 400, error: "Bad Request",…}
error
:
"Bad Request"
exception
:
"#<ActionController::ParameterMissing: param is missing or the value is empty: workout>"
status
:
400"

Within my workouts controller, I have the following:

def workout_params
      params.require(:workout).permit(:plan, :user_id)
    end
end

I'm console logging what I'm passing for data, and it appear that workout is not empty.  Below is what is being recorded in the console:

Object {workout: Object}
    workout: Object
    plan: "pushups"
    user_id: 6

The following is what I have in my JS files for creating workouts:

api.js:
```
const createWorkout = () => {
  return $.ajax({
    url: config.apiOrigin + '/workouts',
    method: 'POST',
      Authorization: 'Token token=' + store.user.token
    },
    data: ''
  })
}
```

events.js:
```
const onCreateWorkout = function (event) {
  event.preventDefault()
  const plan = $('#new-workout-plan').val()
  const userId = store.user.id
  const data = {
    workout: {
      plan: plan,
      user_id: userId
    }
  }
  console.log(data)
  api.createWorkout(data)
    .then(ui.createWorkoutsSuccess)
    .catch(ui.createWorkoutFailure)
}
```

ui.is (only console logging for now):
```
const createWorkoutSuccess = () => {
  console.log('create success')
}

const createWorkoutFailure = () => {
  console.log('create failure')
}
```

Any suggestions on next steps would be appreciated.  Thanks.

**Comment by Jcornmanhomonoff** [comment]
Read your error. It looks like you may be _**missing**_ something

**Comment by payne-chris-r** [comment]
I think Jessica is referring to this line: 
 "#<ActionController::ParameterMissing: param is missing or the value is empty: workout>"

Can you change this line to and post the output?
```diff
const onCreateWorkout = function (event) {
  event.preventDefault()
  const plan = $('#new-workout-plan').val()
  const userId = store.user.id
  const data = {
    workout: {
      plan: plan,
      user_id: userId
    }
  }
-  console.log(data)
+  console.log("data is:", data)
  api.createWorkout(data)
    .then(ui.createWorkoutsSuccess)
    .catch(ui.createWorkoutFailure)
}
```

**Comment by jordanallain** [comment]
any updates @brianedmunds ?

**Comment by brianedmunds** [comment]
Chris - thanks for the added suggestion.

Jordan -  I made the console log changes that Chris suggested, and the output in the console appears to be the same, with the exception that it is now prepended with "data is:":

data is: Object {workout: Object}
  workout: Object
    plan: "pushups"
    user_id: 6

One thing to note is that I have yet to configure my workouts controller for "current_user".  Reason being that I had previously added code for that, but then all of the my requests to the api began failing (not just for create), so I backed that out and am trying to get this working as is, and then will add the code for current user afterwards.  Just wanted to point that out for awareness.

**Comment by brianedmunds** [comment]
After making the above console log change, I did another read over my code and realized what was missing.  I needed to include data during declaration in my api.js file:

const createWorkout = function (data) {

After making that change, I am now able to create from the client successfully.  Didn't notice that yesterday, but it jumped out to me today after having stepped away from the code for a bit.  Closing this out.  Thanks for the help.
